File: Hang-man.py
import random
from random import seed
from random import randint
word = " "
words = ["apple","orange","banana","strawberry","grapes","watermelon","pineapple","pear","mango","cherry"]
s = ['-','-','-','-','-','-','-','-','-','-']
x = 0
def clear():
    global x
    global s
    x = 0
    for i in range(len(s)):
        s[i] = '-'

def select_word():
  global word
  n = random.randint(0,8)
  word = words[n]
def put_letters():
    global s
    global word
    for i in range(len(word)):
         print(s[i] , end = ' ')
    print()
def check_right(c):
    global x
    global word
    global s
    b = False
    for i in range(len(word)):
        if(c == word[i]):
            s[i] = c
            b = True
    if(not b):
        x+=1
def print_man():
     global x
     if(x == 0):
        print(" ___________")
        print(" |          ")
        print(" |         ")
        print("_|______________")
        print()
     elif (x == 1):
            print(" ___________")
            print(" |         0")
            print(" |         ")
            print("_|______________")
            print()
     elif(x == 2):
            print(" ___________")
            print(" |         0")
            print(" |       \\  ")
            print("_|______________")
            print()
     elif(x == 3):
            print(" ___________")
            print(" |         0")
            print(" |       \\ | ")
            print("_|______________")
            print()
     elif(x == 4):
            print(" ___________")
            print(" |         0")
            print(" |       \\ | /")
            print( "_|______________")
            print()
     elif(x == 5):
            print(" ___________")
            print( " |         0")
            print(" |       \\ | /" )
            print(" |         |")
            print("_|______________")
            print()
     elif(x == 6):
            print(" ___________")
            print(" |         0")
            print(" |       \\ | /" )
            print(" |         |")
            print(" |        /  ")
            print("_|______________")
            print()
     elif(x == 7):
            print(" ___________")
            print(" |         0")
            print(" |       \\ | /")
            print(" |         |")
            print(" |        / \\ ")
            print("_|______________")
            print()
def win_lose():
    global s
    global word
    if(x == 7):
        print("YOU LOSE :( !")
        print_man()
        return True
    else:
        for i in range(len(word)):
             if s[i] == '-':
                return False
    print("YOU WIN :)")
    print(word)
    return True
def check_input(l):
    if (ord(l) >= 97 and ord(l) <= 122):
        return True
    return False
def read_input():
    print("GUESS A LETTER: " , end = ' ')
    c = str(input())
    while not check_input(c):
        print("Enter a valid input: ")
        c = str(input())
    return c
def play():
    print("Hangman Game.....")
    print("GUESS a fruit name !!")
    while True:
        print_man()
        put_letters()
        c = read_input()
        check_right(c)
        if(win_lose()):
            break
while True:
    clear()
    select_word()
    play()
    c = input('Play Again [Y/N] ')
    if c not in 'yY':
      break

File: Hangman.cpp
#include <bits/stdc++.h>
using namespace std;
string word;
string words [] = {"apple","orange","banana","strawberry","grapes","watermelon","pineapple","pear","mango","cherry"};
string s = "----------";
int x = 0;
string w[4] = {};
void clear_game()
{
    x = 0;
    s = "----------";
}
void select_word()
{
    srand(time(0));
	int n = rand()% 10;
	word = words[n];
}
void put_letters()
{
    for(int i = 0; i < word.size(); i++)
    {
         cout<<s[i]<<" ";
    }
    cout<<endl;
}
void check_right(char l)
{
    bool b = false;
    for(int i = 0; i < word.size(); i++)
    {
        if(l == word[i])
        {
            s[i] = l;
            b = true;
        }
    }
    if(!b)
        x++;
}
void print(int position)
{

     switch(position)
     {
     case 0:
        cout << " ___________"<<endl;
        cout << " |          "<<endl;
        cout << " |         " <<endl;
        cout << "_|______________"<<endl;
     break;
     case 1:
            cout << " ___________"<<endl;
            cout << " |         0"<<endl;
            cout << " |         " <<endl;
            cout << "_|______________"<<endl;
     break;
     case 2:
            cout << " ___________"<<endl;
            cout << " |         0"<<endl;
            cout << " |       \\  " <<endl;
            cout << "_|______________"<<endl;
     break;
     case 3:       cout << " ___________"<<endl;
            cout << " |         0"<<endl;
            cout << " |       \\ | " <<endl;
            cout << "_|______________"<<endl;
     break;
     case 4:
            cout << " ___________"<<endl;
            cout << " |         0"<<endl;
            cout << " |       \\ | /" <<endl;
            cout << "_|______________"<<endl;
     break;
     case 5:
            cout << " ___________"<<endl;
            cout << " |         0"<<endl;
            cout << " |       \\ | /" <<endl;
            cout << " |         |"<<endl;
            cout << "_|______________"<<endl;
     break;
     case 6:
            cout << " ___________"<<endl;
            cout << " |         0"<<endl;
            cout << " |       \\ | /" <<endl;
            cout << " |         |"<<endl;
            cout << " |        /  "<<endl;
            cout << "_|______________"<<endl;
     break;
     case 7:
            cout << " ___________"<<endl;
            cout << " |         0"<<endl;
            cout << " |       \\ | /" <<endl;
            cout << " |         |"<<endl;
            cout << " |        / \\ "<<endl;
            cout << "_|______________"<<endl;
     }
}

bool win_lose()
{
    if(x == 7)
    {
        cout<<"YOU LOSE :("<<endl;
        print(x);
        return true;
    }
    else
    {
        for(int i = 0; i < word.size(); i++)
        {
             if(s[i] == '-')
                return false;
        }
    }
    cout<<"YOU WIN :)"<<endl;
    cout <<word << endl;
    return true;
}
bool check_input(char l)
{
    if (int(l) >= 97 && int(l) <= 122)
        return true;
    return false;
}
int read_input(string &l)
{
    cout<<"GUESS A LETTER: ";
    cin>> l;
    cout<<endl;
    while (!check_input(l[0]))
    {
        cout<<"Enter a valid input: "<<endl;
        cin >>l;
        cout<<endl;
    }
}
void play()
{
    cout<<"Hangman Game....."<<endl;
    cout<<"GUESS a fruit name !!"<<endl;
    while(true)
    {
        print(x);
        put_letters();
        string i;
        read_input(i);
        check_right(i[0]);
        if(win_lose())
        {

            break;
        }

    }
}
int main()
{
    while(true)
  {
      clear_game();
      select_word();
      play();
      char c;
      cout<<"Play Again [Y/N]";
      cin>>c;
      if(c != 'y' && c!= 'Y')
        break;
 }

}
